### PR TITLE
qa: Adds allowed plugins and updates locked dependencies

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

New errors surfaced by the psalm update have mostly been base-lined so that they can be address in subsequent pulls.

Intending to address some psalm issues for input-filter lib next and wanted to start with a recent version of psalm to avoid things getting messy…

### CODE REVIEW

File: test/Sitemap/LocTest.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
 use function current;
 
 /**
- * @group      Laminas_Validator
+ * @psalm-suppress UnevaluatedCode
```
Comment (Ocramius) [inline test/Sitemap/LocTest.php]: Let's move this to the incomplete test method instead

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/Sitemap/LocTest.php [hunk 2]
```diff
@@ -8,9 +8,7 @@
 
 use function current;
 
-/**
- * @group      Laminas_Validator
- */
+
```
Comment (Ocramius) [inline test/Sitemap/LocTest.php]: Added spacing made CS checks fail, I think

Comment (gsteel) [reply test/Sitemap/LocTest.php]: Doh!

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @gsteel!

**Pull request merged**